var/spack/repos/builtin/packages/amr-wind/package.py

```diff
@@ -98,4 +117,14 @@ def cmake_args(self):
             targets = self.spec.variants["amdgpu_target"].value
             args.append("-DAMReX_AMD_ARCH=" + ";".join(str(x) for x in targets))
 
+        if "+sycl" in self.spec:
+            cmake_options.append(self.define("AMR_WIND_ENABLE_SYCL", True))
+            # SYCL GPU backend only supported with Intel's oneAPI or DPC++ compilers
+            sycl_compatible_compilers = ["dpcpp", "icpx"]
+            if not (os.path.basename(self.compiler.cxx) in sycl_compatible_compilers):
+                raise InstallError(
+                    "AMReX's SYCL GPU Backend requires DPC++ (dpcpp)"
+                    + " or the oneAPI CXX (icpx) compiler."
+                )
```

Comment:
You can turn this into a `requires` directive like:
```python
requires(
        "%dpcpp",
        "%oneapi",
        policy="one_of",
        msg="SYCL GPU backend only supported with Intel's oneAPI or DPC++ compilers",
    )
```
The added benefit is that you'll catch errors in concretization.